twigs/ecr.py

```diff
@@ -0,0 +1,172 @@
+import sys
+import os
+import subprocess
+import logging
+import json
+
+from . import docker
+
+
+g_encoding = None
+
+def set_encoding(encoding):
+    global g_encoding
+    g_encoding = encoding
+
+def get_encoding():
+    global g_encoding
+    return g_encoding
+
+def get_repo_uri(repository, repositoryType):
+
+    if repositoryType == 'private':
+        repo_cmd = "ecr describe-repositories"
+    else:
+        repo_cmd = "ecr-public describe-repositories"
+
+    t_json = run_aws_cmd(repo_cmd)
+
+    if len(t_json) > 0:
+        for i in t_json['repositories']:
+            if i['repositoryName'] == repository:
+                return i['repositoryUri']
+
+def run_aws_cmd(cmd):
+    cmd = 'aws --output json ' + cmd
+    try:
+        cmd_output = subprocess.check_output([cmd], shell=True, stdin=None, stderr=None)
+        cmd_output = cmd_output.decode(get_encoding())
+        ret_json = json.loads(cmd_output)
+
+    #except ClientError as e:
+     #   if e.response['Error']['Code'] == 'RepositoryNotFoundException':
+      #      logging.error("Please make sure that repository exists and you are pointing to correct region",cmd,cmd_output)
+       #     ret_json = { } 
+    except subprocess.CalledProcessError:
+        logging.error("Error running aws  command [%s]", cmd)
+        ret_json = { }
+    except ValueError:
+        logging.error("Error parsing JSON output for aws command [%s]: %s", cmd, cmd_output)
+        ret_json = { }
+    return ret_json
+
+def get_image_name_with_tag(tag,repository,repositoryType):
+
+    if repositoryType == 'private':
+        ilist_cmd = "ecr describe-images --repository-name " + repository
+    else:
+        ilist_cmd = "ecr-public describe-images --repository-name " + repository
+
+    i_json = run_aws_cmd(ilist_cmd)
+
+    if len(i_json['imageDetails']) > 0:
+        i_uri = get_repo_uri(repository,repositoryType)
+
+        for image in i_json['imageDetails']:
+            if tag is None:
+                no_of_tags = len(image['imageTags'])
```

Comment:
If there is no tag specified by the user, then we should use "latest" tag provided one exists, else the most recent tag. Can we update this logic accordingly?

Reply:
Changed the logic to accommodate this.